### ISSUE TITLE
Why are the first few losses all the same?

### ISSUE THREAD

**Issue by Heonjin** [HobbitLong/SupContrast#3]
```
Train: [1][10/196]	BT 0.429 (1.060)	DT 0.000 (0.093)	loss 8.909 (8.925)
Train: [1][20/196]	BT 0.429 (0.744)	DT 0.000 (0.047)	loss 8.909 (8.917)
Train: [1][30/196]	BT 0.426 (0.639)	DT 0.000 (0.031)	loss 8.909 (8.914)
Train: [1][40/196]	BT 0.430 (0.587)	DT 0.000 (0.024)	loss 8.909 (8.913)
Train: [1][50/196]	BT 0.429 (0.555)	DT 0.000 (0.019)	loss 8.909 (8.912)
Train: [1][60/196]	BT 0.431 (0.534)	DT 0.000 (0.016)	loss 8.909 (8.912)
Train: [1][70/196]	BT 0.433 (0.520)	DT 0.000 (0.014)	loss 8.909 (8.911)
Train: [1][80/196]	BT 0.428 (0.508)	DT 0.000 (0.012)	loss 8.909 (8.911)
Train: [1][90/196]	BT 0.431 (0.500)	DT 0.000 (0.011)	loss 8.909 (8.911)
Train: [1][100/196]	BT 0.431 (0.493)	DT 0.000 (0.010)	loss 8.909 (8.911)
Train: [1][110/196]	BT 0.429 (0.487)	DT 0.000 (0.009)	loss 8.909 (8.911)
Train: [1][120/196]	BT 0.430 (0.482)	DT 0.000 (0.008)	loss 8.909 (8.910)
Train: [1][130/196]	BT 0.430 (0.478)	DT 0.000 (0.007)	loss 8.909 (8.910)
Train: [1][140/196]	BT 0.432 (0.475)	DT 0.000 (0.007)	loss 8.909 (8.910)
Train: [1][150/196]	BT 0.429 (0.472)	DT 0.000 (0.006)	loss 8.909 (8.910)
Train: [1][160/196]	BT 0.435 (0.470)	DT 0.000 (0.006)	loss 8.909 (8.910)
Train: [1][170/196]	BT 0.435 (0.467)	DT 0.000 (0.006)	loss 8.909 (8.910)
Train: [1][180/196]	BT 0.428 (0.465)	DT 0.000 (0.005)	loss 8.909 (8.910)
Train: [1][190/196]	BT 0.429 (0.464)	DT 0.000 (0.005)	loss 8.909 (8.910)
epoch 1, total time 92.35
Train: [2][10/196]	BT 0.428 (0.565)	DT 0.000 (0.148)	loss 8.909 (8.909)
Train: [2][20/196]	BT 0.433 (0.498)	DT 0.000 (0.074)	loss 8.909 (8.909)
Train: [2][30/196]	BT 0.430 (0.476)	DT 0.000 (0.050)	loss 8.909 (8.909)
Train: [2][40/196]	BT 0.429 (0.464)	DT 0.000 (0.037)	loss 8.909 (8.909)
Train: [2][50/196]	BT 0.432 (0.458)	DT 0.000 (0.030)	loss 8.909 (8.909)
Train: [2][60/196]	BT 0.437 (0.453)	DT 0.000 (0.025)	loss 8.909 (8.909)
```
Thanks for sharing your code. I run your code of "(2) Supervised Contrastive Learning
Pretraining stage" and it shows the above training losses. I wonder why there are the same first few losses. 

Thanks in advance.

**Comment by HobbitLong** [comment]
/home/yonglong/Desktop/loss.png

**Comment by HobbitLong** [comment]
Here is the loss trajectory.
![loss](https://user-images.githubusercontent.com/29867600/82741333-f39db480-9d1e-11ea-8a73-0942dc70a9a2.png)
I am not sure whether you have adjust the learning rate or not? It seems to me that you are using a smaller batch size than I suggested (1024).

**Comment by Heonjin** [comment]
Oh yes perfect. thanks.

**Comment by Dadatata-JZ** [comment]
@HobbitLong Hi Long, due to the limitation of the computing power, I have to smaller the batch size. I examined by smaller lr however, the loss doesn't drop anymore after 50 epochs (10.* ---> 6.* ). What would be the ideal observation and training strategy you would suggest with smaller batch size? Thank you in advance.
